### PR TITLE
`split-non-commuting` use wire grouping with non-pauli-word observable

### PR DESCRIPTION
**Pull request by astralcai**
**Context:**
The new unified `split_non_commuting` uses `qwc` grouping by default, but it is not supported for non-pauli-word observables.

**Description of the Change:**
Use wire-based grouping if one of the observables is not a pauli word

**Related GitHub Issues:**
Fixes https://github.com/PennyLaneAI/pennylane/issues/4362
[sc-42003]

### CODE REVIEW

**Review by albi3ro — COMMENTED** [review]
Love how simple this fix is :)

Just a couple questions on testing.  Since default.qubit itself can handle measuring projectors with no issue, we should probably also unit test the transform for qscript in, batch out.

Let's also add some tests with hadamards/ products of hadamards.

File: tests/transforms/test_split_non_commuting.py [hunk 1]
```diff
@@ -816,6 +870,23 @@ def circuit():
...
+        @qml.transforms.split_non_commuting
+        @qml.qnode(_dev)
+        def circuit():
+            qml.Hadamard(0)
+            return (
+                qml.expval(qml.Projector([0], wires=[0])),
+                qml.expval(qml.Projector([1], wires=[0])),
+            )
+
+        res = circuit()
```
Comment (albi3ro) [inline tests/transforms/test_split_non_commuting.py:887]:
```suggestion
        with _dev.tracker:
            res = circuit()
        assert _dev.tracker.totals['simulations'] == 2
```

**Review by albi3ro — APPROVED** [review]
Changelog ?

**Comment by astralcai** [comment]
> Changelog ?

Oooops

**Review by PietropaoloFrisoni — APPROVED** [review]
Looks good!

**Pull request merged**